chain/epoch_manager/src/lib.rs

```diff
@@ -1343,26 +1331,22 @@ impl EpochManager {
         &mut self,
         store_update: &mut StoreUpdate,
         epoch_id: &EpochId,
-        epoch_info: EpochInfo,
+        epoch_info: Arc<EpochInfo>,
     ) -> Result<(), EpochError> {
-        store_update
-            .set_ser(ColEpochInfo, epoch_id.as_ref(), &epoch_info)
-            .map_err(EpochError::from)?;
+        store_update.set_ser(ColEpochInfo, epoch_id.as_ref(), &epoch_info)?;
         self.epochs_info.put(epoch_id.clone(), epoch_info);
         Ok(())
     }
 
-    pub fn get_epoch_validator_info(
-        &mut self,
-        epoch_id: &EpochId,
-    ) -> Result<EpochSummary, EpochError> {
+    pub fn get_epoch_validator_info(&self, epoch_id: &EpochId) -> Result<EpochSummary, EpochError> {
         // We don't use cache here since this query happens rarely and only for rpc.
         self.store
-            .get_ser(ColEpochValidatorInfo, epoch_id.as_ref())
-            .map_err(|err| err.into())
-            .and_then(|value| value.ok_or_else(|| EpochError::EpochOutOfBounds(epoch_id.clone())))
+            .get_ser(ColEpochValidatorInfo, epoch_id.as_ref())?
+            .ok_or_else(|| EpochError::EpochOutOfBounds(epoch_id.clone()))
     }
 
+    // NOTE: beware, after calling `save_epoch_validator_info`,
+    // `get_epoch_validator_info` will return stale results.
```

Comment:
nice catch -- we should add a cache to it to avoid this pitfall.

Can you create a github issue and mention it in this comment?